### PR TITLE
Fix issue 2621

### PR DESCRIPTION
**Pull request by ahmedmolawale**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md).
- [x] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [x] My code builds and is tested on a real Android device.
- [x] I confirm that I've run the code locally and everything works as expected.

## What's changed?
Refactored the budget screen and ViewModel to adapt to the current architecture.

### Video
https://github.com/Ivy-Apps/ivy-wallet/assets/12750237/5f6825d9-28ce-470f-8552-b854601cd16e

## Does this PR closes any GitHub Issues?

- Closes [#2621](https://github.com/Ivy-Apps/ivy-wallet/issues/2621)

### CODE REVIEW

File: screen-budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt [hunk 1]
```diff
@@ -45,59 +47,117 @@ class BudgetViewModel @Inject constructor(
...
+            timeRange = getTimeRange(),
+            reorderModalVisible = getReorderModalVisible(),
+            budgetModalData = getBudgetModalData()
+        )
+    }
 
-    private val _baseCurrencyCode =
-        MutableStateFlow(com.ivy.legacy.utils.getDefaultFIATCurrency().currencyCode)
-    val baseCurrencyCode = _baseCurrencyCode.readOnly()
+    override fun onEvent(event: BudgetScreenEvent) {
```
Comment (ILIYANGERMANOV) [inline screen-budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt:81]: Can you place `onEvent` below the `@Composable fun getSomething` functions?

**Review by ILIYANGERMANOV — APPROVED** [review]
Excellent work 👏 Thank you for migrating it! 💯

File: screen-budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt (hunk 1 above)
Comment (ahmedmolawale) [reply screen-budgets/src/main/java/com/ivy/budgets/BudgetViewModel.kt]: Your wish has been granted 😄

**Comment by ILIYANGERMANOV** [comment]
10/10 💯

**Pull request merged**